sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -230,7 +242,9 @@ class FrameLessOffsetWindowFunctionFrame(
     // 7. current row -> z, next selected row -> empty, output: null;
     // ... next selected row is empty, all following return null.
     (current: InternalRow) =>
-      if (nextSelectedRow == EmptyRow) {
+      if (absOffset > input.length) {
```

Comment:
let's not repeat this code in all branches, we can just have a new branch for this
```
if (absOffset > input.length) {
  (current: InternalRow) => _
} else if (ignoreNulls && offset > 0) {
  ...
} ...
```

Reply:
We can't do this. The `input` has not been initialized yet.